Dockerfile

```diff
@@ -1,16 +1,88 @@
-FROM python:3.6
+# =========================================================
+FROM helsinkitest/python-node:3.6-10-slim as staticbuilder
+# ---------------------------------------------------------
+# Stage for building static files for
+# the project. Installs Node as that
+# is required for compiling SCSS files.
+# =========================================================
 
-ENV PYTHONUNBUFFERED 0
+RUN apt-install.sh \
+      libxmlsec1-dev \
+      libxml2-dev \
+      pkg-config \
+      git \
+      curl \
+      libpq-dev \
+      build-essential
 
-RUN apt-get update \
-    && apt-get install -y python-lxml libxmlsec1 libxmlsec1-dev gettext
+WORKDIR /app
 
-RUN mkdir /code
-WORKDIR /code
+COPY requirements.txt /app/requirements.txt
+COPY package.json /app/package.json
+RUN pip install -U pip \
+    && pip install --no-cache-dir  -r /app/requirements.txt
+RUN npm install
 
-COPY . /code
+COPY . /app/
+RUN python manage.py collectstatic --noinput \
+    && python manage.py compilescss
 
-RUN pip install pip==18.0 \
-    && pip install -r /code/requirements.txt \
-    && pip install -r /code/requirements-dev.txt \
-    && pip install prequ==1.4.3
+# ===========================================
+FROM helsinkitest/python:3.6-slim as appbase
+# ===========================================
+
+WORKDIR /app
+
+COPY --chown=appuser:appuser requirements.txt /app/requirements.txt
+COPY --chown=appuser:appuser requirements-prod.txt /app/requirements-prod.txt
+
+# Install main project dependencies and clean up
+# Note that production dependencies are installed here as well since
+# that is the default state of the image and development stages are
+# just extras.
+RUN apt-install.sh \
+      build-essential \
+      libpq-dev \
+      gettext \
+      git \
+      libxmlsec1-dev \
+      libxml2-dev \
+      netcat \
+      pkg-config \
+    && pip install -U pip \
+    && pip install --no-cache-dir  -r /app/requirements.txt \
+    && pip install --no-cache-dir  -r /app/requirements-prod.txt \
+    && apt-cleanup.sh build-essential pkg-config git
+
+COPY docker-entrypoint.sh /app
```

Comment:
No "--chown=appuser:appuser" set. Does the appuser user have sufficient execution rights?

Reply:
Out of scope.